src/xmpp/omemo.c

```diff
@@ -507,11 +542,105 @@ _omemo_receive_devicelist(xmpp_stanza_t* const stanza, void* const userdata)
         }
     }
 
+out:
     omemo_set_device_list(from, device_list);
 
     return 1;
 }
 
+static int
+_omemo_devicelist_publish_result(xmpp_stanza_t* const stanza, void* const userdata)
+{
+    const char* type = xmpp_stanza_get_type(stanza);
+    if (g_strcmp0(type, STANZA_TYPE_ERROR) == 0) {
+        log_error("[OMEMO] Publishing device list failed");
+
+        xmpp_stanza_t *error = xmpp_stanza_get_child_by_name(stanza, "error");
+        if (!error) {
+            log_error("[OMEMO] Missing error element in device list publication result");
+            return 0;
+        }
+
+        xmpp_stanza_t *pubsub_error = xmpp_stanza_get_child_by_ns(error, STANZA_NS_PUBSUB_ERROR);
+        if (!pubsub_error) {
+            // TODO
+            return 0;
+        }
+
+        if (g_strcmp0(xmpp_stanza_get_name(pubsub_error), "precondition-not-met") == 0) {
+            static gboolean reconfigured = false;
```

Comment:
I'm unsure whether this works as intended.

Reply:
we will after the first run always be in `true` but it could be that we switch accounts without restarting profanity. or it could be that we need to republish because while profanity was running another client did something to the devicelist and we need to republish.